cmake/external/gloo.cmake

```diff
@@ -42,7 +42,7 @@ ExternalProject_Add(
     UPDATE_COMMAND        ""
     CONFIGURE_COMMAND     ""
     BUILD_COMMAND         mkdir -p ${GLOO_SOURCE_DIR}/build
-        && cd ${GLOO_SOURCE_DIR}/build && cmake .. && make
+        && cd ${GLOO_SOURCE_DIR}/build && cmake .. -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} && make
```

Comment:
Better use if(WITH_ASCEND) macro to guard the changes. Do not affect the original logic.